Sources/ProjectDescription/TargetDependency.swift

```diff
@@ -73,6 +73,14 @@ public enum TargetDependency: Codable, Hashable {
     ///              e.g. RxSwift
     case packagePlugin(product: String)
 
+    /// Dependency on a Swift Package Manager Swift Macro using Xcode native integration.
+    ///
+    /// - Parameters:
+    ///   - product: The name of the output product. ${PRODUCT_NAME} inside Xcode.
+    ///              e.g. StructBuilder
+    ///
+    case packageMacro(product: String)
```

Comment:
I wonder if something like this would read better:
```swift
case package(product: String, type: PackageType = .library)

// Used as
.package(product: "PackageLibrary"),
.package(product: "PackagePlugin", type: .plugin),
.package(product: "PackagePlugin", type: .macro),
```

Reply:
Yeah it does. I'll add the type, and flag `packagePlugin` as something deprecated that we should drop in the next major version.